### PR TITLE
wayland: Add version 1.23.0

### PR DESCRIPTION
**Pull request by jwillikers**
### Summary
Changes to recipe:  **wayland/1.23.0**

#### Motivation
The latest release of the Wayland library is now available.

#### Details
Add version 1.23.0.
Bump `pkgconf`.
Use a version range for Meson.

Fix the following error from the test package when cross-compiling the Wayland package with Conan 2.5.0.

```
Package wayland-scanner_BUILD was not found in the pkg-config search path.
Perhaps you should add the directory containing `wayland-scanner_BUILD.pc'
to the PKG_CONFIG_PATH environment variable
Package 'wayland-scanner_BUILD' not found
ERROR: wayland/1.23.0 (test package): Error in build() method, line 52
	wayland_scanner = pkg_config.variables["wayland_scanner"]
	ConanException: Error 1 while executing
```

The fix was to set the `PKG_CONFIG_PATH` environment for some reason.

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by AbrilRBS — APPROVED** [review]
Thanks!

File: recipes/wayland/all/test_package/conanfile.py [hunk 1]
```diff
@@ -38,6 +38,10 @@ def generate(self):
         pkg_config_deps.generate()
         virtual_build_env = VirtualBuildEnv(self)
         virtual_build_env.generate()
+        if self._has_build_profile:
+            env = Environment()
+            env.append_path("PKG_CONFIG_PATH", self.generators_folder)
```
Comment (AbrilRBS) [inline recipes/wayland/all/test_package/conanfile.py:43]: I'll need to understand why this is needed before being comfortable approving - maybe @jcar87 has also some insight, he has been fighting with pkgconfig this last week :)

**Comment by jcar87** [comment]
we need details to reproduce this

**Comment by jwillikers** [comment]
> we need details to reproduce this

@jcar87 I'm not sure the log is going to have all the information you need, so let me know if you need anything more.

```
$ conan create . --build missing --update --version 1.23.0
detect_api: Found gcc 14
detect_api: gcc C++ standard library: libstdc++11
detect_api: gcc>=5, using the major as version

======== Exporting recipe to the cache ========
wayland/1.23.0: Exporting package recipe: /var/home/jordan/Projects/conan-center-index/recipes/wayland/all/conanfile.py
wayland/1.23.0: exports: File 'conandata.yml' found. Exporting it...
wayland/1.23.0: Copied 1 '.yml' file: conandata.yml
wayland/1.23.0: Copied 1 '.py' file: conanfile.py
wayland/1.23.0: Exported to cache folder: /home/jordan/.conan2/p/wayla701d46fcf8ec8/e
wayland/1.23.0: Exported: wayland/1.23.0#4c3565fcbd362d795751f792b8be74b3 (2024-07-15 19:29:46 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=armv8
arch.extensions=crc+crypto
arch.processor=cortex-a72.cortex-a53
build_type=Debug
compiler=gcc
compiler.cppstd=20
compiler.libcxx=libstdc++11
compiler.version=11
os=Linux
os.libc=gnu
os.libc.version=2.35
mesa/*:compiler.cppstd=17
[options]
*/*:glx=False
*/*:with_glx=False
*/*:with_ncurses=False
*/*:with_selinux=False
*/*:with_systemd=False
*/*:with_x11=False
*/*:with_xcb=False
*/*:with_xfixes=False
*/*:with_xft=False
*/*:with_xlib=False
*/*:with_xlib_xrender=False
eudev/*:with_kmod=False
libdrm/*:etnaviv=True
libglvnd/*:x11=False
libinput/*:with_libudev=eudev
mesa/*:gallium_driver_etnaviv=True
mesa/*:gallium_vdpau=False
mesa/*:platform_x11=False
mesa/*:with_libudev=eudev
pipewire/*:with_libalsa=False
pipewire/*:with_libudev=False
pipewire/*:with_pulseaudio=False
vulkan-loader/*:with_wsi_xcb=False
vulkan-loader/*:with_wsi_xlib=False
wayland/*:shared=True
weston/*:backend_x11=False
weston/*:with_libudev=eudev
[platform_requires]
opengl/system
[replace_requires]
egl/system: egl/yocto
xkeyboard-config/system: xkeyboard-config/yocto
libdrm/*: libdrm/2.4.109-vendor
libmount/2.39: libmount/2.40.2
libmount/2.39.2: libmount/2.40.2
libselinux/3.5: libselinux/3.6
libudev/system: eudev/3.2.14
util-linux-libuuid/2.39: util-linux-libuuid/2.40.2
util-linux-libuuid/2.39.2: util-linux-libuuid/2.40.2
weston/*: weston/13.0.1-product
[replace_tool_requires]
wayland-protocols/*: wayland-protocols/1.33-vendor
[conf]
tools.build:cflags=['-mcpu=cortex-a72.cortex-a53+crc+crypto', '--sysroot=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux']
tools.build:compiler_executables={'asm': 'aarch64-company-linux-gcc', 'c': 'aarch64-company-linux-gcc', 'cpp': 'aarch64-company-linux-g++'}
tools.build:cxxflags=['-mcpu=cortex-a72.cortex-a53+crc+crypto', '--sysroot=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux']
tools.build:exelinkflags=['-mcpu=cortex-a72.cortex-a53+crc+crypto', '--sysroot=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux']
tools.build:sharedlinkflags=['-mcpu=cortex-a72.cortex-a53+crc+crypto', '--sysroot=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux']
tools.build:sysroot=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux
tools.cmake.cmake_layout:build_folder_vars=['settings.arch', 'settings.compiler', 'options.platform']
tools.cmake.cmaketoolchain:find_package_prefer_config=True
tools.cmake.cmaketoolchain:generator=Ninja
tools.cmake.cmaketoolchain:system_processor=aarch64
tools.cmake.cmaketoolchain:user_toolchain=['/var/home/jordan/.conan2/toolchains/display-4.cmake']
tools.env.virtualenv:powershell=False
tools.meson.mesontoolchain:extra_machine_files=['/var/home/jordan/.conan2/toolchains/sys_root_empty.ini']
tools.system.package_manager:mode=install
tools.system.package_manager:sudo=True
[buildenv]
PATH=+(path)/usr/local/company-product-sdk/sysroots/x86_64-companysdk-linux/sysroots/usr/bin/aarch64-company-linux
PATH=+(path)/usr/local/company-product-sdk/sysroots/x86_64-companysdk-linux/sysroots/usr/bin
CC=aarch64-company-linux-gcc -mcpu=cortex-a72.cortex-a53+crc+crypto --sysroot=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux
CC_LD=gold
CXX=aarch64-company-linux-g++ -mcpu=cortex-a72.cortex-a53+crc+crypto --sysroot=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux
CXX_LD=gold
CPP=aarch64-company-linux-gcc -E -mcpu=cortex-a72.cortex-a53+crc+crypto --sysroot=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux
AS=aarch64-company-linux-as
LD=aarch64-company-linux-ld -mcpu=cortex-a72.cortex-a53+crc+crypto --sysroot=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux
GDB=aarch64-company-linux-gdb
STRIP=aarch64-company-linux-strip
RANLIB=aarch64-company-linux-ranlib
OBJCOPY=aarch64-company-linux-objcopy
OBJDUMP=aarch64-company-linux-objdump
AR=aarch64-company-linux-ar
NM=aarch64-company-linux-nm
M4=m4
TARGET_PREFIX=aarch64-company-linux-
ARCH=arm64
CONFIGURE_FLAGS=--target=aarch64-company-linux --host=aarch64-company-linux --build=x86_64-linux --with-libtool-sysroot=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux
OE_CMAKE_TOOLCHAIN_FILE=/usr/local/company-product-sdk/sysroots/x86_64-companysdk-linux/usr/share/cmake/OEToolchainConfig.cmake
OE_CMAKE_FIND_LIBRARY_CUSTOM_LIB_SUFFIX=`echo lib | sed -e s/lib//`
PKG_CONFIG_SYSROOT_MAP=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux
PKG_CONFIG_PATH=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux/usr/lib/pkgconfig:/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux/usr/share/pkgconfig
PKG_CONFIG_LIBDIR=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux/usr/lib/pkgconfig:/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux/usr/share/pkgconfig
CONFIG_SITE=/usr/local/company-product-sdk/site-config-cortexa72-cortexa53-crypto-company-linux
OECORE_TARGET_ARCH=aarch64
OECORE_TARGET_OS=linux
avahi/*:PKG_CONFIG_SYSROOT_DIR=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux
libxml2/*:PKG_CONFIG_SYSROOT_DIR=/usr/local/company-product-sdk/sysroots/cortexa72-cortexa53-crypto-company-linux

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=20
compiler.libcxx=libstdc++11
compiler.version=14
os=Linux
os.libc=gnu
os.libc.version=2.39
mold/*:compiler.libcxx=libstdc++11
mesa/*:compiler.cppstd=17
[options]
wayland/*:shared=True
[replace_requires]
libmount/2.39: libmount/2.39.2
libselinux/3.5: libselinux/3.6
libsystemd/255.2: libsystemd/255.7
pulseaudio/14.2: pulseaudio/17.0
[conf]
tools.build:compiler_executables={'asm': 'gcc', 'c': 'gcc', 'cpp': 'g++'}
tools.cmake.cmake_layout:build_folder_vars=['settings.arch', 'settings.compiler', 'options.platform']
tools.cmake.cmaketoolchain:find_package_prefer_config=True
tools.cmake.cmaketoolchain:generator=Ninja
tools.cmake.cmaketoolchain:user_toolchain=['/var/home/jordan/.conan2/toolchains/mold.cmake']
tools.env.virtualenv:powershell=False
tools.system.package_manager:mode=install
tools.system.package_manager:sudo=True
[buildenv]
LD=gcc
CC_LD=mold
CXX_LD=mold
avahi/*:LD=!
gdbm/*:LD=!
libalsa/*:LD=!
libffi/*:LD=!
libiconv/*:LD=!
libmount/*:LD=!
libtool/*:LD=!
libva/*:CC_LD=!
flex/*:LD=!
hwloc/*:LD=!
kmod/*:LD=!
pulseaudio/*:CC_LD=!
util-linux-libuuid/*:CC_LD=!


======== Computing dependency graph ========
wayland/1.23.0: Checking remote: conancenter
wayland/1.23.0: Checking remote: conan
libffi/3.4.4: Checking remote: conancenter
libffi/3.4.4: Checking remote: conan
libxml2/2.12.7: Checking remote: conancenter
libxml2/2.12.7: Checking remote: conan
zlib/1.3.1: Checking remote: conancenter
zlib/1.3.1: Checking remote: conan
libiconv/1.17: Checking remote: conancenter
libiconv/1.17: Checking remote: conan
pkgconf/2.1.0: Checking remote: conancenter
pkgconf/2.1.0: Checking remote: conan
meson/1.2.2: Checking remote: conancenter
meson/1.2.2: Checking remote: conan
ninja/1.11.1: Checking remote: conancenter
ninja/1.11.1: Checking remote: conan
expat/2.6.2: Checking remote: conancenter
expat/2.6.2: Checking remote: conan
meson/1.5.0: Checking remote: conancenter
meson/1.5.0: Checking remote: conan
pkgconf/2.2.0: Checking remote: conancenter
pkgconf/2.2.0: Checking remote: conan
Graph root
    cli
Requirements
    expat/2.6.2#2d385d0d50eb5561006a7ff9e356656b - Cache (conan)
    libffi/3.4.4#72cccac6139293b6e97ec9cd676547ff - Cache (conan)
    libiconv/1.17#73fefc1b696e069df90fd1d18aa63edd - Cache (conan)
    libxml2/2.12.7#737a1eba03113d1aa22d67579ced7d5e - Cache (conan)
    wayland/1.23.0#4c3565fcbd362d795751f792b8be74b3 - Not in remote
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2 - Cache (conan)
Build requirements
    expat/2.6.2#2d385d0d50eb5561006a7ff9e356656b - Cache (conan)
    libffi/3.4.4#72cccac6139293b6e97ec9cd676547ff - Cache (conan)
    libiconv/1.17#73fefc1b696e069df90fd1d18aa63edd - Cache (conan)
    libxml2/2.12.7#737a1eba03113d1aa22d67579ced7d5e - Cache (conan)
    meson/1.2.2#04bdfb85d665c82b08a3510aee3ffd19 - Cache (conan)
    meson/1.5.0#1b4a564c208a4ea1362f8f82f1832d60 - Cache (conan)
    ninja/1.11.1#77587f8c8318662ac8e5a7867eb4be21 - Cache (conan)
    pkgconf/2.1.0#27f44583701117b571307cf5b5fe5605 - Cache (conan)
    pkgconf/2.2.0#6462942a22803086372db44689ba825f - Cache (conan)
    wayland/1.23.0#4c3565fcbd362d795751f792b8be74b3 - Not in remote
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2 - Cache (conan)
Resolved version ranges
    expat/[>=2.6.2 <3]: expat/2.6.2
    libxml2/[>=2.12.5 <3]: libxml2/2.12.7
    meson/[>=1.4.0 <2]: meson/1.5.0
    zlib/[>=1.2.11 <2]: zlib/1.3.1

======== Computing necessary packages ========
expat/2.6.2: WARN: Can't update, no package in remote
expat/2.6.2: WARN: Can't update, no package in remote
libffi/3.4.4: WARN: Can't update, no package in remote
libffi/3.4.4: WARN: Can't update, no package in remote
libiconv/1.17: WARN: Can't update, no package in remote
libiconv/1.17: WARN: Can't update, no package in remote
zlib/1.3.1: WARN: Can't update, no package in remote
libxml2/2.12.7: WARN: Can't update, no package in remote
libxml2/2.12.7: WARN: Can't update, no package in remote
wayland/1.23.0: WARN: Can't update, no package in remote
wayland/1.23.0: WARN: Can't update, no package in remote
Requirements
    wayland/1.23.0#4c3565fcbd362d795751f792b8be74b3:d901d5fc31de431ab8165c366f285267446cb0bf#7df8838592442a587c3f7374879a0954 - Cache
Build requirements
Skipped binaries
    expat/2.6.2, libffi/3.4.4, libiconv/1.17, libxml2/2.12.7, zlib/1.3.1, expat/2.6.2, libffi/3.4.4, libiconv/1.17, libxml2/2.12.7, meson/1.2.2, meson/1.5.0, ninja/1.11.1, pkgconf/2.1.0, pkgconf/2.2.0, wayland/1.23.0, zlib/1.3.1

======== Installing packages ========
wayland/1.23.0: Already installed! (1 of 1)
WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN: deprecated:     'env_info' used in: wayland/1.23.0

======== Launching test_package ========

======== Computing dependency graph ========
Graph root
    wayland/1.23.0 (test package): /var/home/jordan/Projects/conan-center-index/recipes/wayland/all/test_package/conanfile.py
Requirements
    expat/2.6.2#2d385d0d50eb5561006a7ff9e356656b - Cache
    libffi/3.4.4#72cccac6139293b6e97ec9cd676547ff - Cache
    libiconv/1.17#73fefc1b696e069df90fd1d18aa63edd - Cache
    libxml2/2.12.7#737a1eba03113d1aa22d67579ced7d5e - Cache
    wayland/1.23.0#4c3565fcbd362d795751f792b8be74b3 - Cache
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2 - Cache
Build requirements
    expat/2.6.2#2d385d0d50eb5561006a7ff9e356656b - Cache
    libffi/3.4.4#72cccac6139293b6e97ec9cd676547ff - Cache
    libiconv/1.17#73fefc1b696e069df90fd1d18aa63edd - Cache
    libxml2/2.12.7#737a1eba03113d1aa22d67579ced7d5e - Cache
    meson/1.2.2#04bdfb85d665c82b08a3510aee3ffd19 - Cache
    meson/1.5.0#1b4a564c208a4ea1362f8f82f1832d60 - Cache
    ninja/1.11.1#77587f8c8318662ac8e5a7867eb4be21 - Cache
    pkgconf/2.1.0#27f44583701117b571307cf5b5fe5605 - Cache
    pkgconf/2.2.0#6462942a22803086372db44689ba825f - Cache
    wayland/1.23.0#4c3565fcbd362d795751f792b8be74b3 - Cache
    zlib/1.3.1#f52e03ae3d251dec704634230cd806a2 - Cache

======== Computing necessary packages ========
Requirements
    wayland/1.23.0#4c3565fcbd362d795751f792b8be74b3:d901d5fc31de431ab8165c366f285267446cb0bf#7df8838592442a587c3f7374879a0954 - Cache
Build requirements
    pkgconf/2.1.0#27f44583701117b571307cf5b5fe5605:cea2c4e68b521119c0b7a567bf8a9248c3a623f5#ffd89d29b0c8ce283e61b8de0efc6df8 - Cache
    wayland/1.23.0#4c3565fcbd362d795751f792b8be74b3:eac3b51ab61eb0e2e073f21a44d175603c320b83#caf3d69c3c62a46984bf8a40c4b37d65 - Cache
Skipped binaries
    expat/2.6.2, libffi/3.4.4, libiconv/1.17, libxml2/2.12.7, zlib/1.3.1, expat/2.6.2, libffi/3.4.4, libiconv/1.17, libxml2/2.12.7, meson/1.2.2, meson/1.5.0, ninja/1.11.1, pkgconf/2.2.0, zlib/1.3.1

======== Installing packages ========
pkgconf/2.1.0: Already installed! (1 of 3)
pkgconf/2.1.0: WARN: The use of 'unix_path_legacy_compat' is deprecated in Conan 2.0 and does not perform path conversions. This is retained for compatibility with Conan 1.x and will be removed in a future version.
wayland/1.23.0: Already installed! (2 of 3)
wayland/1.23.0: Already installed! (3 of 3)
WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN: deprecated:     'env_info' used in: pkgconf/2.1.0, wayland/1.23.0

======== Testing the package ========
Removing previously existing 'test_package' build folder: /var/home/jordan/Projects/conan-center-index/recipes/wayland/all/test_package/build/armv8-gcc/Debug
wayland/1.23.0 (test package): Test package build: build/armv8-gcc/Debug
wayland/1.23.0 (test package): Test package build folder: /var/home/jordan/Projects/conan-center-index/recipes/wayland/all/test_package/build/armv8-gcc/Debug
wayland/1.23.0 (test package): Calling generate()
wayland/1.23.0 (test package): Generators folder: /var/home/jordan/Projects/conan-center-index/recipes/wayland/all/test_package/build/armv8-gcc/Debug/generators
wayland/1.23.0 (test package): CMakeToolchain generated: conan_toolchain.cmake
wayland/1.23.0 (test package): CMakeToolchain generated: /var/home/jordan/Projects/conan-center-index/recipes/wayland/all/test_package/build/armv8-gcc/Debug/generators/CMakePresets.json
wayland/1.23.0 (test package): CMakeToolchain generated: /var/home/jordan/Projects/conan-center-index/recipes/wayland/all/test_package/CMakeUserPresets.json
wayland/1.23.0 (test package): CMakeDeps necessary find_package() and targets for your CMakeLists.txt
    find_package(wayland)
    target_link_libraries(... wayland::wayland)
wayland/1.23.0 (test package): WARN: PkgConfigDeps.build_context_suffix attribute has been deprecated. Use PkgConfigDeps.build_context_folder instead.
wayland/1.23.0 (test package): Generating aggregated env files
wayland/1.23.0 (test package): Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']

======== Testing the package: Building ========
wayland/1.23.0 (test package): Calling build()
wayland/1.23.0 (test package): Running CMake.configure()
wayland/1.23.0 (test package): RUN: cmake -G "Ninja" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/var/home/jordan/Projects/conan-center-index/recipes/wayland/all/test_package" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Debug" "/var/home/jordan/Projects/conan-center-index/recipes/wayland/all/test_package"
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Using Conan toolchain: /var/home/jordan/Projects/conan-center-index/recipes/wayland/all/test_package/build/armv8-gcc/Debug/generators/conan_toolchain.cmake
-- Conan toolchain: C++ Standard 20 with extensions OFF
-- The C compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/local/company-product-sdk/sysroots/x86_64-companysdk-linux/usr/bin/aarch64-company-linux/aarch64-company-linux-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Conan: Component target declared 'wayland::wayland-scanner'
-- Conan: Component target declared 'wayland::wayland-server'
-- Conan: Component target declared 'wayland::wayland-client'
-- Conan: Component target declared 'wayland::wayland-egl-backend'
-- Conan: Component target declared 'wayland::wayland-cursor'
-- Conan: Component target declared 'wayland::wayland-egl'
-- Conan: Target declared 'wayland::wayland'
-- Configuring done (0.3s)
-- Generating done (0.0s)
-- Build files have been written to: /var/home/jordan/Projects/conan-center-index/recipes/wayland/all/test_package/build/armv8-gcc/Debug

wayland/1.23.0 (test package): Running CMake.build()
wayland/1.23.0 (test package): RUN: cmake --build "/var/home/jordan/Projects/conan-center-index/recipes/wayland/all/test_package/build/armv8-gcc/Debug" -- -j20
[1/2] Building C object CMakeFiles/test_package.dir/test_package.c.o
[2/2] Linking C executable test_package

Package wayland-scanner_BUILD was not found in the pkg-config search path.
Perhaps you should add the directory containing `wayland-scanner_BUILD.pc'
to the PKG_CONFIG_PATH environment variable
Package 'wayland-scanner_BUILD', required by 'virtual:world', not found
ERROR: wayland/1.23.0 (test package): Error in build() method, line 49
	wayland_scanner = pkg_config.variables["wayland_scanner"]
	ConanException: Error 1 while executing
```

**Comment by danimtb** [comment]
Hi @jwillikers,

Thanks for the report and the logs. Since the issue is not related to the new version, I will open a separate pull-request to move those changes forward. That way we can work on a proper fix for that on an isolated PR.

Regarding the error and checking the logs, it is a bit weird that the PKG_CONFIG_PATH is not properly set, as that was included long time ago in Conan 1.55 https://github.com/conan-io/conan/pull/12513.

We need to understand how this is affecting the build. Could you verify the value of that PKG_CONFIG_PATH variable in the test_package build? Maybe the wayland recipe has to set something else for consumers?

**Comment by danimtb** [comment]
@jwillikers take a look at the comment at https://github.com/conan-io/conan-center-index/pull/25983#issuecomment-2489101603

The test package was simplified and additionally we did not found any issue with crossbuilding.
So, I will close this PR for now but please report any issue you may have and we'll help you with that.

Thanks!

**Pull request closed without merging**